File: src/HttPlaceholder.Models/StubConditionsModel.cs
using System.Collections.Generic;
using YamlDotNet.Serialization;

namespace HttPlaceholder.Models
{
   public class StubConditionsModel
   {
      [YamlMember(Alias = "method")]
      public string Method { get; set; }

      [YamlMember(Alias = "url")]
      public StubUrlConditionModel Url { get; set; }

      [YamlMember(Alias = "body")]
      public IEnumerable<string> Body { get; set; }

      [YamlMember(Alias = "form")]
      public IEnumerable<StubFormModel> Form { get; set; }

      [YamlMember(Alias = "headers")]
      public IDictionary<string, string> Headers { get; set; }

      [YamlMember(Alias = "xpath")]
      public IEnumerable<StubXpathModel> Xpath { get; set; }

      [YamlMember(Alias = "jsonPath")]
      public IEnumerable<string> JsonPath { get; set; }

      [YamlMember(Alias = "basicAuthentication")]
      public StubBasicAuthenticationModel BasicAuthentication { get; set; }

      [YamlMember(Alias = "clientIp")]
      public string ClientIp { get; set; }

      [YamlMember(Alias = "host")]
      public string Host { get; set; }
   }
}

File: gui/src/domain/stub/stub-conditions-model.ts
import type { StubUrlConditionModel } from "@/domain/stub/stub-url-condition-model";
import type { StubFormModel } from "@/domain/stub/stub-form-model";
import type { StubXpathModel } from "@/domain/stub/stub-xpath-model";
import type { StubBasicAuthenticationModel } from "@/domain/stub/stub-basic-authentication-model";
import type { StubConditionsScenarioModel } from "@/domain/stub/stub-conditions-scenario-model";

export interface StubConditionsModel {
  method?: string | string[];
  url?: StubUrlConditionModel;
  body?: any[];
  form?: StubFormModel[];
  headers?: any;
  xpath?: StubXpathModel[];
  jsonPath?: any[];
  basicAuthentication?: StubBasicAuthenticationModel;
  clientIp?: string;
  host?: any;
  json?: any;
  scenario?: StubConditionsScenarioModel;
}
